src/Mvc/Extensions.ApiDescription.Client/src/build/Microsoft.Extensions.ApiDescription.Client.targets

```diff
@@ -0,0 +1,149 @@
+    </ItemGroup>
+  </Target>
+
+  <!-- OpenApiReference support. -->
+
+  <Target Name="_GetMetadataForOpenApiReferences" Condition="'@(OpenApiReference)' != ''">
+    <ItemGroup>
+      <_Temporary Remove="@(_Temporary)" />
+    </ItemGroup>
+
+    <GetFileReferenceMetadata Inputs="@(OpenApiReference)"
+        Extension="$(DefaultLanguageSourceExtension)"
+        Namespace="$(RootNamespace)"
+        OutputDirectory="$(OpenApiDefaultOutputDirectory)">
+      <Output TaskParameter="Outputs" ItemName="_Temporary" />
+    </GetFileReferenceMetadata>
+
+    <ItemGroup>
+      <OpenApiReference Remove="@(OpenApiReference)" />
+      <OpenApiReference Include="@(_Temporary)" />
+      <_Temporary Remove="@(_Temporary)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_GetCurrentOpenApiReference">
+    <GetCurrentItems Input="$(GeneratorMetadata)">
+      <Output TaskParameter="Outputs" ItemName="CurrentOpenApiReference" />
+    </GetCurrentItems>
+  </Target>
+
+  <Target Name="_InnerGenerateOpenApiReferenceCode" DependsOnTargets="_GetCurrentOpenApiReference;$(GeneratorTarget)" />
+
+  <Target Name="_GenerateOpenApiReferenceCode"
+      Condition="$(OpenApiGenerateAtDesignTime) OR ('$(DesignTimeBuild)' != 'true' AND '$(BuildingProject)' == 'true')"
+      Inputs="@(OpenApiReference)"
+      Outputs="%(OutputPath)">
+    <MSBuild BuildInParallel="$(BuildInParallel)"
+        Projects="$(MSBuildProjectFullPath)"
+        Properties="GeneratorTargetPath=%(OpenApiReference.OutputPath);GeneratorTarget=Generate%(CodeGenerator);GeneratorMetadata=%(SerializedMetadata)"
+        RemoveProperties="TargetFrameworks"
+        Targets="_InnerGenerateOpenApiReferenceCode" />
+  </Target>
+
+  <Target Name="_CreateCompileItemsForOpenApiReferences" Condition="'@(OpenApiReference)' != ''">
+    <ItemGroup>
+      <_Files Remove="@(_Files)" />
+      <_Files Include="@(OpenApiReference -> '%(OutputPath)')"
+          Condition="$([System.IO.File]::Exists('%(OpenApiReference.OutputPath)'))">
+        <OutputPathExtension>$([System.IO.Path]::GetExtension('%(OpenApiReference.OutputPath)'))</OutputPathExtension>
+      </_Files>
+      <_Directories Remove="@(_Directories)" />
+      <_Directories Include="@(OpenApiReference -> '%(OutputPath)')"
+          Condition="Exists('%(OpenApiReference.OutputPath)') AND ! $([System.IO.File]::Exists('%(OpenApiReference.OutputPath)'))" />
+
+      <!-- If OutputPath is a file, add it directly to relevant items. -->
+      <TypeScriptCompile Include="@(_Files)"
+          Exclude="@(TypeScriptCompile)"
+          Condition="'%(_Files.OutputPathExtension)' == '.ts' OR '%(_Files.OutputPathExtension)' == '.tsx'">
+        <SourceDocument>%(_Files.FullPath)</SourceDocument>
+      </TypeScriptCompile>
+
+      <Compile Include="@(_Files)"
+          Exclude="@(Compile)"
+          Condition="'$(DefaultLanguageSourceExtension)' != '.ts' AND '%(_Files.OutputPathExtension)' == '$(DefaultLanguageSourceExtension)'">
```

Comment:
Have we tested this with F# projects? If not, should we limit this to 'Condition="'$(Language)'=='C#'"`?

Reply:
This condition is correct because compiled languages other than TypeScript are not mixed into other projects. That is, this is about the _exception_ to normal mono-language projects.